portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,32 +26,44 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
 
 
 /**
-  * Servlet that returns some example content. 
+  * Servlet that returns comments in database. 
   * Note: We first added the Gson library dependency to pom.xml.
   */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  int maxNumComments = 1;
   ArrayList<String> msglist = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    maxNumComments = Integer.parseInt(request.getParameter("numComment"));
     // Create the query and prepared query to load comment entities from database
     Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
-    // Add all comment contents to the msgList  
+    // Add all comment contents to the msglist  
     ArrayList<String> msglist = new ArrayList<String>();
     for (Entity comment:results.asIterable()) {
         String commentMsg = (String)comment.getProperty("content");
         msglist.add(commentMsg);
     }
+    Collections.shuffle(msglist);
```

Comment:
I shuffle it just because I want to refresh comments every time the user click show comments. As it is explained in the walkthroughs: "Now that the user can change how many comments should be displayed, you'll need to give them a way to refresh the comments section when they make a selection." Now every time a user click "show the comments" button, they can see different ones.

Reply:
I'm not sure if this makes sense. Since they are already in chronological order, they should be displayed in chronological order. By shuffling don't you mess up the order. If they select the same number of max comments, they should see the same comments. I think that quote is referring to when they change the number of max comments they should see more/less comments.